packages/application-shell/src/components/application-shell/application-shell.tsx

```diff
@@ -141,336 +141,345 @@ export const RestrictedApplication = <
-          const hasUnauthorizedError = getHasUnauthorizedError(
-            error.graphQLErrors
-          );
-          const hasUserBeenDeletedError = getHasUserBeenDeletedError(
-            error.graphQLErrors
-          );
-
-          if (hasUnauthorizedError || hasUserBeenDeletedError) {
-            let logoutReason:
-              | typeof LOGOUT_REASONS[keyof typeof LOGOUT_REASONS]
-              | undefined;
-            if (hasUnauthorizedError)
-              logoutReason = LOGOUT_REASONS.UNAUTHORIZED;
-            else if (hasUserBeenDeletedError)
-              logoutReason = LOGOUT_REASONS.DELETED;
-            return (
-              <Redirector to="logout" queryParams={{ reason: logoutReason }} />
+) => {
+  const location = useLocation();
+  return (
+    <FetchUser>
+      {({ isLoading: isLoadingUser, user, error }) => {
+        if (error) {
+          // In case there is an unauthorized error, we redirect to the login page
+          if (error.graphQLErrors && Array.isArray(error.graphQLErrors)) {
+            const hasUnauthorizedError = getHasUnauthorizedError(
+              error.graphQLErrors
+            );
+            const hasUserBeenDeletedError = getHasUserBeenDeletedError(
+              error.graphQLErrors
             );
+
+            if (hasUnauthorizedError || hasUserBeenDeletedError) {
+              let logoutReason:
+                | typeof LOGOUT_REASONS[keyof typeof LOGOUT_REASONS]
+                | undefined;
+              if (hasUnauthorizedError)
+                logoutReason = LOGOUT_REASONS.UNAUTHORIZED;
+              else if (hasUserBeenDeletedError)
+                logoutReason = LOGOUT_REASONS.DELETED;
+              return (
+                <Redirector
+                  to="logout"
+                  queryParams={{ reason: logoutReason }}
+                />
+              );
+            }
           }
+          // Since we do not know the locale of the user, we pick it from the
+          // user's browser to attempt to match the language for the correct translations.
+          const userLocale = getBrowserLocale(window);
+          return (
+            <AsyncLocaleData
+              locale={userLocale}
+              applicationMessages={props.applicationMessages}
+            >
+              {({ locale, messages }) => {
+                reportErrorToSentry(error, {});
+                return (
+                  <ConfigureIntlProvider locale={locale} messages={messages}>
+                    <ErrorApologizer />
+                  </ConfigureIntlProvider>
+                );
+              }}
+            </AsyncLocaleData>
+          );
         }
-        // Since we do not know the locale of the user, we pick it from the
-        // user's browser to attempt to match the language for the correct translations.
-        const userLocale = getBrowserLocale(window);
+
+        const projectKeyFromUrl = selectProjectKeyFromUrl(location.pathname);
```

Comment:
And here we read the project key from the URL. Note that we're passing the pathname from the router location.